### ISSUE TITLE
gopls: automated issue report (crash)

### ISSUE THREAD

**Issue by hzhu212** [golang/vscode-go#2403]
gopls version: v0.9.2 (go1.18)
gopls flags: 
update flags: proxy
extension version: 0.35.1
go version: 1.18
environment: Visual Studio Code darwin
initialization error: undefined
issue timestamp: Fri, 12 Aug 2022 07:20:24 GMT
restart history:
Fri, 12 Aug 2022 07:13:48 GMT: activation (enabled: true)

ATTENTION: PLEASE PROVIDE THE DETAILS REQUESTED BELOW.

Describe what you observed.

Just open any .go file, there will be an alert message saying "Connection to server got closed. Server will not be restarted."

Please attach the stack trace from the crash.
A window with the error message should have popped up in the lower half of your screen.
Please copy the stack trace and error messages from that window and paste it in this issue.

```
[Info  - 3:26:43 PM] 2022/08/12 15:26:43 go env for /Users/xxx/workspace/demo
(root /Users/xxx/workspace/demo)
(go version go version go1.18 darwin/amd64)
(valid build configuration = false)
(build flags: [])
GOPATH=/Users/xxx/go
GOCACHE=/Users/xxx/Library/Caches/go-build
GOWORK=
GOINSECURE=
GOMOD=/dev/null
GONOPROXY=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOPRIVATE=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOFLAGS=
GOSUMDB=sum.golang.google.cn
GO111MODULE=on
GOROOT=/usr/local/Cellar/go/1.18/libexec
GOPROXY=https://goproxy.xxxx.org|https://goproxy.cn|direct
GONOSUMDB=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOMODCACHE=/Users/xxx/go/pkg/mod


[Info  - 3:26:43 PM] 2022/08/12 15:26:43 go/packages.Load #1
	snapshot=0
	directory=/Users/xxx/workspace/demo
	query=[./ builtin]
	packages=2

[Info  - 3:26:43 PM] 2022/08/12 15:26:43 go/packages.Load #1: updating metadata for 0 packages

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 go/packages.Load #3
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/test.go]
	packages=1

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 go/packages.Load #3
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/test.go]

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 go/packages.Load #3: updating metadata for 1 packages

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 reloadOrphanedFiles reloading
	query=[file:///Users/xxx/workspace/demo/go-demo/guess.go]

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 go/packages.Load #4
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/guess.go]
	packages=1

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 go/packages.Load #4
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/guess.go]

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 go/packages.Load #4: updating metadata for 43 packages

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 go/packages.Load #6
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/guess.go]
	packages=1

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 go/packages.Load #6
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/guess.go]

[Info  - 3:26:44 PM] 2022/08/12 15:26:44 go/packages.Load #6: updating metadata for 0 packages

panic: invalid start/end token.Pos

goroutine 210 [running]:
golang.org/x/tools/internal/span.NewRange(...)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/span/token.go:33
golang.org/x/tools/internal/lsp/cache.(*snapshot).applyCriticalErrorToFiles(0xc00069b900, {0x1c86138, 0xc000884090}, {0x1a6cc66, 0x193}, {0xc0027cbac0?, 0x2, 0xc000f79570?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:383 +0x58a
golang.org/x/tools/internal/lsp/cache.(*snapshot).workspaceLayoutError(0xc00069b900, {0x1c86138, 0xc000884090})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:330 +0x20b
golang.org/x/tools/internal/lsp/cache.(*snapshot).GetCriticalError(0xc00069b900, {0x1c86138, 0xc000884090})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/snapshot.go:1391 +0x2b6
golang.org/x/tools/internal/lsp.(*Server).diagnose(0xc0002cc240, {0x1c86138, 0xc0004eabd0}, {0x1c8e970, 0xc00069b900}, 0x0)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:264 +0xecd
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshot(0xc0002cc240, {0x1c8e970, 0xc00069b900}, {0xc0002eb400, 0x1, 0x1}, 0x0?)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:153 +0x2f1
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots.func1({0x1c8e970?, 0xc00069b900?}, {0xc0002eb400?, 0x12acd4a?, 0x0?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:130 +0x85
created by golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:128 +0x90
[Info  - 3:26:44 PM] 
[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go env for /Users/xxx/workspace/demo
(root /Users/xxx/workspace/demo)
(go version go version go1.18 darwin/amd64)
(valid build configuration = false)
(build flags: [])
GO111MODULE=on
GOWORK=
GOINSECURE=
GONOPROXY=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOMOD=/dev/null
GONOSUMDB=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOCACHE=/Users/xxx/Library/Caches/go-build
GOMODCACHE=/Users/xxx/go/pkg/mod
GOPATH=/Users/xxx/go
GOROOT=/usr/local/Cellar/go/1.18/libexec
GOPROXY=https://goproxy.xxxx.org|https://goproxy.cn|direct
GOPRIVATE=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOFLAGS=
GOSUMDB=sum.golang.google.cn


[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #1
	snapshot=0
	directory=/Users/xxx/workspace/demo
	query=[./ builtin]
	packages=2

[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #1: updating metadata for 0 packages

[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #3
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/test.go]

[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #3: updating metadata for 1 packages

[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #3
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/test.go]
	packages=1

[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #4
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/guess.go]
	packages=1

[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #4
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/guess.go]

[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #4: updating metadata for 43 packages

panic: invalid start/end token.Pos

goroutine 81 [running]:
golang.org/x/tools/internal/span.NewRange(...)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/span/token.go:33
golang.org/x/tools/internal/lsp/cache.(*snapshot).applyCriticalErrorToFiles(0xc0002a6f00, {0x1c86138, 0xc000f9f1d0}, {0x1a6cc66, 0x193}, {0xc00215f100?, 0x2, 0xc001c37570?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:383 +0x58a
golang.org/x/tools/internal/lsp/cache.(*snapshot).workspaceLayoutError(0xc0002a6f00, {0x1c86138, 0xc000f9f1d0})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:330 +0x20b
golang.org/x/tools/internal/lsp/cache.(*snapshot).GetCriticalError(0xc0002a6f00, {0x1c86138, 0xc000f9f1d0})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/snapshot.go:1391 +0x2b6
golang.org/x/tools/internal/lsp.(*Server).diagnose(0xc0001fe7e0, {0x1c86138, 0xc00094a0c0}, {0x1c8e970, 0xc0002a6f00}, 0x0)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:264 +0xecd
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshot(0xc0001fe7e0, {0x1c8e970, 0xc0002a6f00}, {0xc000458ef0, 0x1, 0x1}, 0x1?)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:153 +0x2f1
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots.func1({0x1c8e970?, 0xc0002a6f00?}, {0xc000458ef0?, 0x0?, 0x0?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:130 +0x85
created by golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:128 +0x90
[Info  - 3:26:45 PM] 
[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go env for /Users/xxx/workspace/demo
(root /Users/xxx/workspace/demo)
(go version go version go1.18 darwin/amd64)
(valid build configuration = false)
(build flags: [])
GONOSUMDB=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOPATH=/Users/xxx/go
GO111MODULE=on
GOPRIVATE=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOMODCACHE=/Users/xxx/go/pkg/mod
GONOPROXY=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOMOD=/dev/null
GOSUMDB=sum.golang.google.cn
GOROOT=/usr/local/Cellar/go/1.18/libexec
GOINSECURE=
GOPROXY=https://goproxy.xxxx.org|https://goproxy.cn|direct
GOFLAGS=
GOWORK=
GOCACHE=/Users/xxx/Library/Caches/go-build


[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #1
	snapshot=0
	directory=/Users/xxx/workspace/demo
	query=[./ builtin]
	packages=2

[Info  - 3:26:45 PM] 2022/08/12 15:26:45 go/packages.Load #1: updating metadata for 0 packages

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #3
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/test.go]
	packages=1

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #3
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/test.go]

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #3: updating metadata for 1 packages

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #4
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/guess.go]

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #4
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/guess.go]
	packages=1

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #4: updating metadata for 43 packages

panic: invalid start/end token.Pos

goroutine 159 [running]:
golang.org/x/tools/internal/span.NewRange(...)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/span/token.go:33
golang.org/x/tools/internal/lsp/cache.(*snapshot).applyCriticalErrorToFiles(0xc0001cb2c0, {0x1c86138, 0xc00150a360}, {0x1a6cc66, 0x193}, {0xc001df9560?, 0x2, 0xc000627570?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:383 +0x58a
golang.org/x/tools/internal/lsp/cache.(*snapshot).workspaceLayoutError(0xc0001cb2c0, {0x1c86138, 0xc00150a360})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:330 +0x20b
golang.org/x/tools/internal/lsp/cache.(*snapshot).GetCriticalError(0xc0001cb2c0, {0x1c86138, 0xc00150a360})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/snapshot.go:1391 +0x2b6
golang.org/x/tools/internal/lsp.(*Server).diagnose(0xc00080c5a0, {0x1c86138, 0xc0006b4120}, {0x1c8e970, 0xc0001cb2c0}, 0x0)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:264 +0xecd
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshot(0xc00080c5a0, {0x1c8e970, 0xc0001cb2c0}, {0xc0005b29f0, 0x1, 0x1}, 0x1?)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:153 +0x2f1
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots.func1({0x1c8e970?, 0xc0001cb2c0?}, {0xc0005b29f0?, 0x1?, 0x8?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:130 +0x85
created by golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:128 +0x90
[Info  - 3:26:46 PM] 
[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go env for /Users/xxx/workspace/demo
(root /Users/xxx/workspace/demo)
(go version go version go1.18 darwin/amd64)
(valid build configuration = false)
(build flags: [])
GOMOD=/dev/null
GONOSUMDB=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOCACHE=/Users/xxx/Library/Caches/go-build
GOINSECURE=
GOPATH=/Users/xxx/go
GOFLAGS=
GOSUMDB=sum.golang.google.cn
GOROOT=/usr/local/Cellar/go/1.18/libexec
GOPRIVATE=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOPROXY=https://goproxy.xxxx.org|https://goproxy.cn|direct
GOWORK=
GO111MODULE=on
GONOPROXY=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOMODCACHE=/Users/xxx/go/pkg/mod


[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #1: updating metadata for 0 packages

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #1
	snapshot=0
	directory=/Users/xxx/workspace/demo
	query=[./ builtin]
	packages=2

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #3
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/test.go]
	packages=1

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #3
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/test.go]

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #3: updating metadata for 1 packages

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #4
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/guess.go]
	packages=1

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #4
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/guess.go]

[Info  - 3:26:46 PM] 2022/08/12 15:26:46 go/packages.Load #4: updating metadata for 43 packages

panic: invalid start/end token.Pos

goroutine 154 [running]:
golang.org/x/tools/internal/span.NewRange(...)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/span/token.go:33
golang.org/x/tools/internal/lsp/cache.(*snapshot).applyCriticalErrorToFiles(0xc0001977c0, {0x1c86138, 0xc0005224e0}, {0x1a6cc66, 0x193}, {0xc0023605a0?, 0x2, 0xc0015f3570?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:383 +0x58a
golang.org/x/tools/internal/lsp/cache.(*snapshot).workspaceLayoutError(0xc0001977c0, {0x1c86138, 0xc0005224e0})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:330 +0x20b
golang.org/x/tools/internal/lsp/cache.(*snapshot).GetCriticalError(0xc0001977c0, {0x1c86138, 0xc0005224e0})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/snapshot.go:1391 +0x2b6
golang.org/x/tools/internal/lsp.(*Server).diagnose(0xc0003e8120, {0x1c86138, 0xc0007ca5d0}, {0x1c8e970, 0xc0001977c0}, 0x0)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:264 +0xecd
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshot(0xc0003e8120, {0x1c8e970, 0xc0001977c0}, {0xc0002665c0, 0x1, 0x1}, 0x38?)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:153 +0x2f1
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots.func1({0x1c8e970?, 0xc0001977c0?}, {0xc0002665c0?, 0xc0002831f0?, 0x0?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:130 +0x85
created by golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:128 +0x90
[Info  - 3:26:46 PM] 
[Info  - 3:26:47 PM] 2022/08/12 15:26:47 go env for /Users/xxx/workspace/demo
(root /Users/xxx/workspace/demo)
(go version go version go1.18 darwin/amd64)
(valid build configuration = false)
(build flags: [])
GOCACHE=/Users/xxx/Library/Caches/go-build
GOINSECURE=
GOMOD=/dev/null
GOPATH=/Users/xxx/go
GOMODCACHE=/Users/xxx/go/pkg/mod
GOWORK=
GONOSUMDB=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOROOT=/usr/local/Cellar/go/1.18/libexec
GOPROXY=https://goproxy.xxxx.org|https://goproxy.cn|direct
GOSUMDB=sum.golang.google.cn
GO111MODULE=on
GOPRIVATE=*.xxxx.org,*.everphoto.cn,git.smartisan.com
GOFLAGS=
GONOPROXY=*.xxxx.org,*.everphoto.cn,git.smartisan.com


[Info  - 3:26:47 PM] 2022/08/12 15:26:47 go/packages.Load #1
	snapshot=0
	directory=/Users/xxx/workspace/demo
	query=[./ builtin]
	packages=2

[Info  - 3:26:47 PM] 2022/08/12 15:26:47 go/packages.Load #1: updating metadata for 0 packages

[Info  - 3:26:47 PM] 2022/08/12 15:26:47 go/packages.Load #4
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/test.go]

[Info  - 3:26:47 PM] 2022/08/12 15:26:47 go/packages.Load #4
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/test.go]
	packages=1

[Info  - 3:26:47 PM] 2022/08/12 15:26:47 go/packages.Load #4: updating metadata for 1 packages

[Info  - 3:26:47 PM] 2022/08/12 15:26:47 go/packages.Load #3
	snapshot=2
	directory=/Users/xxx/workspace/demo
	query=[file=/Users/xxx/workspace/demo/go-demo/guess.go]
	packages=1

[Info  - 3:26:47 PM] 2022/08/12 15:26:47 go/packages.Load #3
	snapshot=2
	package="command-line-arguments"
	files=[/Users/xxx/workspace/demo/go-demo/guess.go]

[Info  - 3:26:47 PM] 2022/08/12 15:26:47 go/packages.Load #3: updating metadata for 43 packages

panic: invalid start/end token.Pos

goroutine 166 [running]:
golang.org/x/tools/internal/span.NewRange(...)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/span/token.go:33
golang.org/x/tools/internal/lsp/cache.(*snapshot).applyCriticalErrorToFiles(0xc00014d2c0, {0x1c86138, 0xc001c50090}, {0x1a6cc66, 0x193}, {0xc0017657c0?, 0x2, 0xc0009eb570?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:383 +0x58a
golang.org/x/tools/internal/lsp/cache.(*snapshot).workspaceLayoutError(0xc00014d2c0, {0x1c86138, 0xc001c50090})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/load.go:330 +0x20b
golang.org/x/tools/internal/lsp/cache.(*snapshot).GetCriticalError(0xc00014d2c0, {0x1c86138, 0xc001c50090})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/cache/snapshot.go:1391 +0x2b6
golang.org/x/tools/internal/lsp.(*Server).diagnose(0xc0002f25a0, {0x1c86138, 0xc000114ed0}, {0x1c8e970, 0xc00014d2c0}, 0x0)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:264 +0xecd
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshot(0xc0002f25a0, {0x1c8e970, 0xc00014d2c0}, {0xc000610b80, 0x1, 0x1}, 0x0?)
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:153 +0x2f1
golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots.func1({0x1c8e970?, 0xc00014d2c0?}, {0xc000610b80?, 0x0?, 0x0?})
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:130 +0x85
created by golang.org/x/tools/internal/lsp.(*Server).diagnoseSnapshots
	/Users/xxx/go/pkg/mod/golang.org/x/tools@v0.1.13-0.20220808185556-e884e0ef6dcd/internal/lsp/diagnostics.go:128 +0x90
[Error - 3:26:47 PM] Connection to server got closed. Server will not be restarted.
```

Failed to auto-collect gopls trace: no gopls log.

OPTIONAL: If you would like to share more information, you can attach your complete gopls logs.

NOTE: THESE MAY CONTAIN SENSITIVE INFORMATION ABOUT YOUR CODEBASE.
DO NOT SHARE LOGS IF YOU ARE WORKING IN A PRIVATE REPOSITORY.

<OPTIONAL: ATTACH LOGS HERE>

**Comment by findleyr** [comment]
Thank you for the report! This was fixed in gopls@v0.9.3, which was released yesterday. Please upgrade gopls (either from the command line, or via "Go: Install/Update Tools".

Sorry for the crash!

**Comment by hzhu212** [comment]
> Thank you for the report! This was fixed in gopls@v0.9.3, which was released yesterday. Please upgrade gopls (either from the command line, or via "Go: Install/Update Tools".
> 
> Sorry for the crash!

Great! It worked after upgrading gopls, thanks for your support!
